### ISSUE TITLE
Question : disableDefaultSettingPanel: true, how to provide a plugin settings panel?

### ISSUE THREAD

**Issue by aderici** [alibaba/lowcode-engine#1348]
Hi,

I noticed the disableDefaultSettingPanel option which disables the default one. I tried copying original settings panel and converting it into a plugin but it throws errors. 

Is this the intended use of disableDefaultSettingPanel, providing custom settings panels as a plugin?

Can you provide a reference implementation with the standard panel but as a plugin? 

Thanks and best regards,

**Comment by aderici** [comment]
@JackLian thank you so much for offering help. 

Here's the repo : https://github.com/aderici/lowcode-demo-custom-settingspanel/commit/f0bfae309441cd9fbe20bb248a80fd19e595a130

I'm using demo-general.

And the error message : 

<img width="761" alt="image" src="https://user-images.githubusercontent.com/6830693/207118801-50f55386-daf5-489c-9895-befe31c36ffc.png">

**Comment by JackLian** [comment]
looks like you can not use settingsPane from @alilc/lowcode-editor-skeleton directly， since there are some apis can not be used outside of engine, which is why this settingsPane is not a standalone plugin yet.

**Comment by aderici** [comment]
@JackLian the primary reason is i want to translate texts into english language. Forexample in settings pane it opens by default with a non-i18n text 

<img width="372" alt="image" src="https://user-images.githubusercontent.com/6830693/207249981-24537470-3361-4885-bfc2-12c9ff6c2f0d.png">


Also the default outline pane has a search box which has a hard-coded non-i18n placeholder : 

<img width="351" alt="image" src="https://user-images.githubusercontent.com/6830693/207250082-8196b2cb-1204-4411-a17c-2851cfc665f4.png">

**Comment by JackLian** [comment]
got it, I will handle that lately， maybe  in 2-4 week, is that OK？

**Comment by aderici** [comment]
@JackLian Yes thank you! I was able to get all the rest translated including setters because they can be used as plugins. 


I will study materials part in the meanwhile,

**Comment by JackLian** [comment]
I`ll provide a demo of how to do i18n in plugins using standard api tomorrow

**Comment by JackLian** [comment]
actually about plugin`s i18n demo, you can check this out 
https://github.com/alibaba/lowcode-plugins/blob/d98b9efdcef5ff9f1b9f233ee1fff72c27ddc78d/packages/plugin-zh-en/src/index.tsx#L1

this show`s how we recommend to do i18n for plugin, and this depends on lowcode-engine@1.0.18 ( right now 1.0.18-beta.4 is the latest)

**Comment by aderici** [comment]
@JackLian thanks, a great example. 

If we could get the engine part to support english then people like me can customize their individual plugins without having to worry about the core.

I'll now try to have an asset library that supports english in component names and default structures when you drag drop them into the designer area

**Comment by huxingyi1997** [comment]
> 
Any plan to release this? And there're some case only has Chinese now in plugins.
For example, '搜索组件' in plugin-components-pane
![Screen Shot 2022-12-26 at 3 50 34 PM](https://user-images.githubusercontent.com/58174093/209521179-338fa7ec-f501-4e91-acb5-ccbf0ae9aa86.png)
, 保存schema in plugin-schema
![Screen Shot 2022-12-26 at 3 50 51 PM](https://user-images.githubusercontent.com/58174093/209521450-9263ba07-83c9-4f9a-91be-9060a3eabb95.png)

**Comment by alvarto** [comment]
> maybe in two weeks
> 
> > 
> 
> Any plan to release this? And there're some case only has Chinese now in plugins. For example, '搜索组件' in plugin-components-pane ![Screen Shot 2022-12-26 at 3 50 34 PM](https://user-images.githubusercontent.com/58174093/209521179-338fa7ec-f501-4e91-acb5-ccbf0ae9aa86.png) , 保存schema in plugin-schema ![Screen Shot 2022-12-26 at 3 50 51 PM](https://user-images.githubusercontent.com/58174093/209521450-9263ba07-83c9-4f9a-91be-9060a3eabb95.png)

Let's track this in [this issue](https://github.com/alibaba/lowcode-engine/issues/1412)
